.github/pull_request_template.md

```diff
@@ -0,0 +1,41 @@
+### Pull Request (PR) Title
+
+Things the PR does go here
+- these can be bullet
+- points like this
+- if you like
+
+### Pre-PR Checklist:
+
+Have you
+- [ ] Updated NUbook if necessary (add link to NUbook PR here)
+- [ ] Added/Updated tests for your changes, including regression tests for bug fixes
+- [ ] Updated relevant module READMEs
+- [ ] Added relevant labels to the PR
+
```

Comment:
```suggestion
- [ ] Ensured that the PR builds successfully (or issues are identified here)
- [ ] [Formatted any documents edited](https://nubook.nubots.net/guides/general/git#formatting-and-buildkite-checks)
- [ ] [Added any required directives in documentation](https://nubook.nubots.net/guides/general/documentation#extra-formatting)

```

I agree with what Cameron suggested earlier this doesn't need to be a tutorial for how to create a PR and acts only to confirms any necessary info exists and cleans it up consistently.

Just figured particularly with newer members this info and links in the checklist may be useful (also if we mix up our documentation style in future or otherwise change formatting etc linking over to the NUbook page will help keep this up to date and make it easy to jump over to in order to check the current standards)

Reply:
I personally don't think we should add things which are caught by CI like formatting and builds to the template - I reckon we should avoid making this template too large, lest people not bother reading it (it's probably larger than I would like already :( )
I reckon that the directives are probably worth adding though :+1: good idea.

Regarding difficulty labels, I had originally intended them to be for issues and things which you were looking for other people to contribute on. In this example, although this PR isn't "hard" to contribute to, it's not something I especially want someone to jump in and contribute to because they saw it's "easy" for them to help with, which was the original intention of those labels. If you think those labels could be improved with a different conceptualisation though, I am all ears.